packages/edit-site/src/components/page-pages/index.js

```diff
@@ -32,12 +33,14 @@ export default function PagePages() {
 		pageSize: PAGE_SIZE_VALUES[ 0 ],
 	} );
 	// Request post statuses to get the proper labels.
-	const [ postStatuses, setPostStatuses ] = useState( EMPTY_ARRAY );
-	useEffect( () => {
-		apiFetch( {
-			path: '/wp/v2/statuses',
-		} ).then( setPostStatuses );
-	}, [] );
+	const { records: statuses } = useEntityRecords( 'root', 'status' );
+	const postStatuses =
+		statuses === null
+			? EMPTY_OBJECT
+			: statuses.reduce( ( acc, status ) => {
+					acc[ status.slug ] = status.name;
+					return acc;
+			  }, EMPTY_OBJECT );
```

Comment:
@oandregal: This breaks a few things in the component — even though right now that's asymptomatic. :)

The cause is the use of `EMPTY_OBJECT` inside the `Array#reduce` call, because the reducer will mutate that initial value. This means that:

* `postStatuses` will always have the exact same object reference, regardless of whether `statuses === null`, and thus any subsequent logic in the component that relies on reference changes (e.g. `useMemo( ..., [ postStatuses ] )` will not fire correctly.
* `EMPTY_OBJECT` will become non-empty as soon as `statuses` resolves.

Proof of bug:

```js
const EMPTY_OBJECT = {};

function getPostStatuses( rawStatuses ) {
	return rawStatuses === null
		? EMPTY_OBJECT
		: rawStatuses.reduce( ( acc, status ) => {
				acc[ status.slug ] = status.name;
				return acc;
		  }, EMPTY_OBJECT );
}

const before = getPostStatuses( null );
const after = getPostStatuses( [ { slug: 'draft', name: 'Draft' } ] );

console.log( 'Has the reference changed?', before !== after );
console.log(
	'Is EMPTY_OBJECT empty?',
	Object.keys( EMPTY_OBJECT ).length === 0
);
```

The immediate fix is to not use `EMPTY_OBJECT` in the mutating context:

```diff
	const postStatuses =
		statuses === null
			? EMPTY_OBJECT
			: statuses.reduce( ( acc, status ) => {
					acc[ status.slug ] = status.name;
					return acc;
-			  }, EMPTY_OBJECT );
+			  }, {} );
```

But of course this means that `postStatuses` will always have a new reference even when nothing has changed, so the solution is to use `useMemo`. But, while we are here, I recommend against using `Array#reduce` in favour of clearer higher-level constructs:

```js
	const postStatuses = useMemo(
		() =>
			statuses === null
				? EMPTY_OBJECT
				: Object.fromEntries(
						statuses.map( ( { slug, name } ) => [ slug, name ] )
				  ),
		[ statuses ]
	);
```

I'll turn this into a PR, but I wanted to give you my full explanation first!

Reply:
-> https://github.com/WordPress/gutenberg/pull/55314